spinalcordtoolbox/reports/qc.py

```diff
@@ -419,12 +420,16 @@ def update_description_file(self, dimension):
         }
         logger.debug('Description file: %s', self.qc_params.qc_results)
         results = []
-        if os.path.isfile(self.qc_params.qc_results):
-            results = json.load(open(self.qc_params.qc_results, 'r'))
-        results.append(output)
-        json.dump(results, open(self.qc_params.qc_results, "w"), indent=2)
+        with sct.open_with_exclusive_lock(self.qc_params.qc_results, 'r+') as lck_qc_file:
+            if os.path.getsize(self.qc_params.qc_results) is not 0:
```

Comment:
It turns out that cpython is interning small integers, but it's not a good idea to use `is` for anything but `None, True, False, Ellipsis`.